bundles/org.openhab.ui/web/src/components/widgets/widget-mixin.js

```diff
@@ -68,6 +69,24 @@ export default {
       return true
     }
   },
+  mounted () {
+    if (this.config && this.config.stylesheet) {
+      this.css_uid = 'scoped-' + Math.random().toString(36).substr(2, 9)
```

Comment:
- `this.css_uid` should be initialized (to null) in `data ()` and also be camelCase instead of snake_case (i.e. `cssUid`).
- to generate a random ID `this.$f7.utils.id()` is used throughout the app, maybe use that here as well?

Reply:
- right, sorry, been sick last week (is that a valid excuse for all these "consistency faults"? 😆 )
- hm, I should have looked deeper into this. I thought that `id()` generates a uuid, which I thought is unnecessarily long in this case (well, eventually it wouldn't have mattered either). Checking the docs I now realize that by default it creates a 10 chars hex hash, which is about the same. So, will adjust, of course!

Reply:
ah, and regarding `data ()`:
I meanwhile read on multiple websites that it's not best practice to put variables in data() that don't need to be reactive, and that one should avoid simply initializing all "component variables" there. That's actually the reason I didn't put it there.
It's up to you how to handle this. Eventually, I think it doesn't hurt much, it just creates some getters and setters that are never called...

Reply:
> * `this.css_uid` should be initialized (to null) in `data ()` and also be camelCase instead of snake_case (i.e. `cssUid`).
>
> * to generate a random ID `this.$f7.utils.id()` is used throughout the app, maybe use that here as well?

fixed. what about `data ()` ?

Reply:
> what about data () ?

Your argument above seems fair enough to me.